internal/cmd/plugin/logs/pretty/pretty.go

```diff
@@ -0,0 +1,242 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pretty
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/cloudnative-pg/machinery/pkg/stringset"
+	"github.com/logrusorgru/aurora/v4"
+	"github.com/spf13/cobra"
+
+	"github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin"
+)
+
+type prettyCmd struct {
+	loggers  *stringset.Data
+	pods     *stringset.Data
+	minLevel LogLevel
+}
+
+// NewCmd creates a new `kubectl cnpg logs pretty` command
+func NewCmd() *cobra.Command {
+	var loggers, pods []string
+	bf := prettyCmd{}
+
+	cmd := &cobra.Command{
+		Use:   "pretty",
+		Short: "Prettify CNPG logs",
+		Long:  "Reads CNPG logs from the standard input and pretty-print them for human consumption",
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return plugin.CompleteClusters(cmd.Context(), args, toComplete), cobra.ShellCompDirectiveNoFileComp
+		},
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			bf.loggers = stringset.From(loggers)
+			bf.pods = stringset.From(pods)
+
+			recordChannel := make(chan logRecord)
+			recordGroupsChannel := make(chan []logRecord)
+
+			var wait sync.WaitGroup
+
+			wait.Add(1)
+			go func() {
+				bf.decode(cmd.Context(), os.Stdin, recordChannel)
+				wait.Done()
+			}()
+
+			wait.Add(1)
+			go func() {
+				bf.group(cmd.Context(), recordChannel, recordGroupsChannel)
+				wait.Done()
+			}()
+
+			wait.Add(1)
+			go func() {
+				bf.write(cmd.Context(), recordGroupsChannel, os.Stdout)
+				wait.Done()
+			}()
+
+			wait.Wait()
+			return nil
+		},
+	}
+
+	cmd.Flags().StringSliceVar(&loggers, "loggers", nil,
+		"The list of loggers to receive. Defaults to all.")
+	cmd.Flags().StringSliceVar(&pods, "pods", nil,
+		"The list of pods to receive from. Defaults to all.")
+	cmd.Flags().Var(&bf.minLevel, "min-level",
+		`Hides the messages whose log level is less important than the specified one.
+Should be empty or one of error, warning, info, debug, or trace.`)
+
+	return cmd
+}
+
+// decode progressively decode the logs
+func (bf *prettyCmd) decode(ctx context.Context, reader io.Reader, recordChannel chan<- logRecord) {
+	scanner := bufio.NewScanner(reader)
+
+	for scanner.Scan() {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+
+		record, err := newLogRecordFromBytes(scanner.Bytes())
+		if err != nil {
+			_, _ = fmt.Fprintln(
+				os.Stderr,
+				aurora.Red(fmt.Sprintf("JSON syntax error (%s)", err.Error())),
+				scanner.Text())
+			continue
+		}
+
+		record.normalize()
+
+		if !bf.isRecordRelevant(record) {
+			continue
+		}
+
+		recordChannel <- *record
+	}
+
+	close(recordChannel)
+}
+
+// group transforms a stream of logs in a stream of log groups
+func (bf *prettyCmd) group(ctx context.Context, logChannel <-chan logRecord, groupChannel chan<- []logRecord) {
```

Comment:
I'm guessing the main reason to have this is to be able to sort by timestamp within  each group of 200 logs.
Is it worth it? I mean, is it a bad assumption that the process producing the logs, be that `stern` or `Kubectl logs` or `Kubectl cnpg logs`, will return logs in chronological order?

Reply:
The fixed 200 is not my worry here, but the complexity added.
I guess we could `cat` several files with logs, and then this would be a great feature.
But if the intent is to pipe after `stern` or `Kubectl logs`, I think this is overkill, no?

Reply:
I think that, without sorting, logs are very difficult to interpret. The differentiation brought by CNPG is the concept of cluster, not single pods.

This is an attempt to go in that direction and it is a good enough implementation for now, enabling users to test it widely, before investing more in the feature.

My 2 cents here.

Reply:
Makes sense. Thanks

Reply:
Adding a command line flag for the group size is a good idea. I'm adding it.

Reply:
Done!